### PR TITLE
deps: upgrade next.js v15.5.18

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
Closes ENG-1536

Tested by performing a checkout (until payment)

# Pull Request Checklist

- [x] Test the changes locally to ensure they work as expected.
- [x] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [x] Include new tests for any new functionality or significant changes.
- [x] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: apps/storefront/src/app/[locale]/layout.tsx [hunk 1]
```diff
@@ -32,7 +31,7 @@ export default async function LocaleLayout({
   params,
 }: {
   children: ReactNode;
-  params: Promise<{ locale: SupportedLocale }>;
+  params: Promise<{ locale: string }>;
```
Comment (NyanKiyoshi) [inline apps/storefront/src/app/[locale]/layout.tsx:34]: Fixes:

```
8.357 storefront:build: > storefront@1.0.0 build /app/apps/storefront
8.357 storefront:build: > next build
8.357 storefront:build:
9.248 storefront:build:    ▲ Next.js 15.5.18
9.249 storefront:build:    - Experiments (use with caution):
9.249 storefront:build:      · clientTraceMetadata
9.249 storefront:build:
9.868 storefront:build:    Creating an optimized production build ...
46.46 storefront:build: <w> [webpack.cache.PackFileCacheStrategy] Serializing big strings (128kiB) impacts deserialization performance (consider using Buffer instead and decode when needed)
46.48 storefront:build: <w> [webpack.cache.PackFileCacheStrategy] Serializing big strings (318kiB) impacts deserialization performance (consider using Buffer instead and decode when needed)
48.88 storefront:build: Browserslist: browsers data (caniuse-lite) is 12 months old. Please run:
48.88 storefront:build:   npx update-browserslist-db@latest
48.88 storefront:build:   Why you should do it regularly: https://github.com/browserslist/update-db#readme
57.74 storefront:build:  ✓ Compiled successfully in 48s
57.75 storefront:build:    Linting and checking validity of types ...
74.89 storefront:build: Failed to compile.
74.90 storefront:build:
74.90 storefront:build: Type error: Type 'typeof import("/app/apps/storefront/src/app/[locale]/layout")' does not satisfy the constraint 'LayoutConfig<"/[locale]">'.
74.90 storefront:build:   Types of property 'default' are incompatible.
74.90 storefront:build:     Type '({ children, params, }: { children: ReactNode; params: Promise<{ locale: "en-GB"; }>; }) => Promise<Element>' is not assignable to type 'ComponentType<LayoutProps<"/[locale]">> | ((props: LayoutProps<"/[locale]">) => void | ReactNode | Promise<void> | Promise<...>)'.
74.90 storefront:build:       Type '({ children, params, }: { children: ReactNode; params: Promise<{ locale: "en-GB"; }>; }) => Promise<Element>' is not assignable to type 'FunctionComponent<LayoutProps<"/[locale]">>'.
74.90 storefront:build:         Types of parameters '__0' and 'props' are incompatible.
74.90 storefront:build:           Type 'LayoutProps<"/[locale]">' is not assignable to type '{ children: ReactNode; params: Promise<{ locale: "en-GB"; }>; }'.
74.90 storefront:build:             Types of property 'params' are incompatible.
74.90 storefront:build:               Type 'Promise<{ locale: string; }>' is not assignable to type 'Promise<{ locale: "en-GB"; }>'.
74.90 storefront:build:                 Type '{ locale: string; }' is not assignable to type '{ locale: "en-GB"; }'.
74.90 storefront:build:                   Types of property 'locale' are incompatible.
74.90 storefront:build:                     Type 'string' is not assignable to type '"en-GB"'.
74.90 storefront:build:
74.92 storefront:build: Next.js build worker exited with code: 1 and signal: null
75.00 storefront:build:  ELIFECYCLE  Command failed with exit code 1.
75.02 storefront:build: ERROR: command finished with error: command (/app/apps/storefront) /usr/local/bin/pnpm run build exited (1)
75.03 storefront#build: command (/app/apps/storefront) /usr/local/bin/pnpm run build exited (1)
75.04
75.04  Tasks:    1 successful, 2 total
75.04 Cached:    0 cached, 2 total
75.04   Time:    1m13.452s
75.04 Failed:    storefront#build
75.04
75.07  ERROR  run failed: command  exited (1)
75.14  ELIFECYCLE  Command failed with exit code 1.
75.19  ELIFECYCLE  Command failed with exit code 1.
```

**Review by lkostrowski — APPROVED** [review]
